GameServerLib/GameObjects/AttackableUnits/AI/Champion.cs

```diff
@@ -356,26 +356,13 @@ public override void Die(IAttackableUnit killer)
         public override void OnCollision(IGameObject collider, bool isTerrain = false)
         {
             // TODO: Pathfinding should be responsible for pathing around units so collisions with other units never occur (or at least very little).
-            // Collisions only occur between buildings.
-            if (MovementParameters != null || !(collider is IObjBuilding || collider is IBaseTurret || isTerrain == true))
+            // As a result, collisions for Champions should only occur between buildings and terrain.
+            if (!(collider is IBaseTurret || isTerrain == true))
             {
                 return;
```

Comment:
I don't really understand this part (I probably forgot something here). If we have a chain of inheritance Champion<->ObjAiBase <-> AttackableUnit <-> ..., I expect each to handle its priorities and then to continue propagating the collision to the next ones. Why isn't that the case with the Champion class colliding with let's say a minion? Do we have down-casting to AttackableUnit when checking this collision?